### PR TITLE
Move privilege e2e test to common

### PR DESCRIPTION
**Pull request by Jefftree**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Move security privilege test from `test/e2e_node` to `test/e2e/common`. This allows us to promote the test to a Conformance test.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/common/security_context.go [hunk 1]
```diff
@@ -272,6 +272,19 @@ var _ = framework.KubeDescribe("Security Context", func() {
 				e2elog.Failf("unprivileged container shouldn't be able to create dummy device")
 			}
 		})
+
+		ginkgo.It("should run the container as privileged when true [NodeFeature:HostAccess]", func() {
```
Comment (johnbelamaric) [inline test/e2e/common/security_context.go]: Add [LinuxOnly] tag too

Comment (Jefftree) [reply test/e2e/common/security_context.go]: ack

**Comment by johnbelamaric** [comment]
I wonder what the difference is between `test/e2e/common/security_context.go` and `test/e2e/node/security_context.go` is? @tallclair ?

**Comment by Jefftree** [comment]
This was confirmed offline with @tallclair. `e2e_node` tests are lower level and are meant for testing the Kubelet code on a custom host environment. Certain functionalities like the scheduler are also omitted in `e2e_node` tests. For this specific test, since the unprivileged test is already conformant and the privileged test does not depend on anything specific from `e2e_node`, I think it's fine to move over from `e2e_node` to `e2e/common`.

**Review by tallclair — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Comment by tallclair** [comment]
Actually, this test is sort of redundant (has some slight differences) with https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/privileged.go

Can you add a TODO to merge those tests?

**Comment by Jefftree** [comment]
> Actually, this test is sort of redundant (has some slight differences) with https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/privileged.go
> 
> Can you add a TODO to merge those tests?

ack. Added a TODO note in `privileged.go`

**Pull request merged**